File: backend_springboot/src/main/java/com/polideportivo/backend_springboot/domain/model/Grafica.java
package com.polideportivo.backend_springboot.domain.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "graficas")
public class Grafica {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "seccion", nullable = false)
    private String seccion;

    @Column(name = "nivel", nullable = false)
    private Integer nivel;

    @Column(name = "año", nullable = false)
    private Integer año;

    @Column(name = "mes", nullable = false)
    private Integer mes;

    @ManyToOne
    @JoinColumn(name = "profile_id", nullable = false)
    private Profile profile;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;
}

File: backend_laravel/app/Models/Grafica.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Grafica extends Model
{
    use HasFactory;

   
    protected $table = 'graficas';

   
    protected $fillable = [
        'año',
        'mes',
        'seccion',
        'nivel',
        'profile_id',
    ];

    
    public function profile()
    {
        return $this->belongsTo(Profile::class);
    }
}
